components/kyma-environment-broker/internal/runtime/handler.go

```diff
@@ -313,6 +305,34 @@ func (h *Handler) setRuntimeOptionalAttributes(instance internal.Instance, dto *
 	return nil
 }
 
+func determineKymaVersion(pOprs []internal.ProvisioningOperation, uOprs []internal.UpgradeKymaOperation) string {
+	kymaVersion := ""
+	kymaVersionSetAt := time.Time{}
+
+	// Set kyma version from the last provisioning operation
+	if len(pOprs) != 0 {
+		kymaVersion = pOprs[0].RuntimeVersion.Version
+		kymaVersionSetAt = pOprs[0].CreatedAt
+	}
+
+	// Take the last upgrade kyma operation which
+	//   - is not dry-run
+	//   - is created after the last provisioning operation
+	//   - has the kyma version set
+	//   - the has been processed, i.e. not pending, canceling or canceled
```

Comment:
nit: I would drop "the" from "the has been..."

Reply:
Fixed.